src/ReferenceWidget.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+	<div>
+		<div v-if="reference && hasCustomWidget" class="widget-custom">
+			<div ref="customWidget" />
+		</div>
+
+		<div v-else-if="noAccess" class="widget-access">
+			<p>You may not have access to this reference!</p>
```

Comment:
Needs translation or a slot/prop so apps can translate it?

Reply:
I'd actually go and just not show it here, for special widgets like file where access permission is more critical, the apps can provide a custom error message with some more context about sharing then anyways.